src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -266,11 +277,18 @@ private void CommitAndPersistNode(Hash256? address, ref TreePath path, in NodeCo
         static void ThrowUnknownHash(TrieNode node) => throw new TrieStoreException($"The hash of {node} should be known at the time of committing.");
     }
 
-    private int GetNodeShardIdx(in TreePath path, Hash256 hash) =>
+    private int GetNodeShardIdx(in TreePath path, Hash256 hash)
+    {
         // When enabled, the shard have dictionaries for tracking past path hash also.
         // So the same path need to be in the same shard for the remove logic to work.
-        // Using the address first byte however, causes very uneven distribution. So the path is used.
-        _livePruningEnabled ? path.Path.Bytes[0] : hash.Bytes[0];
+        uint baseSize = _livePruningEnabled
+            ? BinaryPrimitives.ReadUInt32BigEndian(path.Path.Bytes[..4])
+            : BinaryPrimitives.ReadUInt32BigEndian(hash.Bytes[..4]);
```

Comment:
```suggestion
        uint baseSize = (uint)_livePruningEnabled
            ? path..GetHashCode()
            : hash..GetHashCode();
```

Reply:
This was a change I was looking to do in https://github.com/NethermindEth/nethermind/pull/7406 to reduce contention, but saw you already had PR open in the area 😅